python/tvm/driver/tvmc/compiler.py

```diff
@@ -0,0 +1,407 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Provides support to compile networks both AOT and JIT.
+"""
+import argparse
+import logging
+import tarfile
+from pathlib import Path
+
+import tvm
+from tvm import autotvm
+from tvm import relay
+from tvm._ffi.runtime_ctypes import TVMContext
+from tvm.contrib import cc
+from tvm.contrib import util
+from tvm.relay.op.contrib import get_pattern_table
+
+from . import common, frontends
+from .main import register_parser
+
+# A dictionary of target aliases to simplify the command lines provided by end users
+TARGET_ALIASES = {
+    "aarch64": "llvm -device=arm_cpu -mtriple=aarch64-linux-gnu -mattr=+neon"
+}
+
+#  A list of valid targets (including aliases) to be used in "--target"
+VALID_TARGETS = ["aarch64", "llvm"]
+
+DEFAULT_TARGET = "llvm"
+DUMP_FORMATS = ["relay", "ll", "asm"]
+
+
+def parse_target(targets_str):
+    """ Parsing function for comma separated target syntax. """
+    targets = targets_str.split(",")
+    for target in targets:
+        if target not in VALID_TARGETS:
+            raise argparse.ArgumentTypeError(f"unrecognized target: {target}")
+    return targets
+
+
+@register_parser
+def add_compile_parser(subparsers):
+    """ Include parser for 'compile' subcommand """
+
+    parser = subparsers.add_parser("compile", help="compile a model")
+    parser.set_defaults(func=drive_compile)
+    parser.add_argument(
+        "--cross-compiler",
+        default="",
+        help="the cross compiler to use to generate target libraries",
+    )
+    parser.add_argument(
+        "--dump-codegen", default="", choices=DUMP_FORMATS, help="dump generated code"
+    )
+    parser.add_argument(
+        "--language",
```

Comment:
The naming is confusing. Based on its use cases, it is the file format of the input model file. I won't say "keras", "onnx", or "pytorch" are languages.

Reply:
I agree that `--language` doesn't seem the best option here. Discussing with @mbaret, we came up with the idea of changing it to `--model-format`. What do you think?

Reply:
`--model-format` sounds reasonable. Ideally, this could be an optional argument. When `--model-format` is given then we directly use it; otherwise we infer the format from the model file extension (e.g., `.onnx`) and throw an error when failed. This approach is commonly used by many packages already.

Reply:
Yes, that is the case. We try guessing the framework from a list of known extensions per frontend. So in my next update, this will be renamed.